[14.0][MIG] cooperator: migration to 14.0

huguesdk opened the pull request:
## description

migrate cooperator to odoo 14.0

## odoo task

[t8755](https://gestion.coopiteasy.be/web#id=8755&model=project.task&view_type=form)

houssine78 committed:
[IMP] allow cancel when sub request is in waiting state

houssine78 committed:
[FIX] get mail template should been done on newly created sub request

houssine78 committed:
[FIX] remove uneeded buggy assignation

houssine78 committed:
[FIX] assign created partner to self if created

houssine78 committed:
[FIX] fix wrong attribute name

houssine78 committed:
[IMP] remove commented code

robinkeunen committed:
[FIX] fix demo data journal id

robinkeunen committed:
[IMP] emc: hide share lines menu

robinkeunen committed:
[IMP] name group_by_cooperator_type and filter def consistency

robinkeunen committed:
[FIX] emc: get_share_type returns ids

robinkeunen committed:
[FIX] emc: cooperator account demo data

robinkeunen committed:
[FIX] subscription request validate button visibility + test

houssine78 committed:
[FIX] take new use case into account: sub req for non cooperator

houssine78 committed:
[MIG] add _get_report_base_filename method

remytms committed:
[FIX] emc: footer for report

houssine78 committed:
[TYPO] fix typo

houssine78 committed:
[IMP] improve code

houssine78 committed:
[REFACT] refact functions

remytms committed:
[FIX] emc: Report header hiding destinator's name

mclaeysb committed:
[ADD] emc: add the date of certificate generation on certificate

robinkeunen committed:
[IMP] easy_my_coop_online_payment: not installable

robinkeunen committed:
[REF] easy_my_coop: todos

robinkeunen committed:
Revert 4f01fde "[REF] easy_my_coop: todos"

robinkeunen committed:
[IMP] emc: demo data skips control

raneq committed:
Pypi setup

raneq committed:
Translate to catalan the "Become cooperator" form.

Also translate the email templates related to share movements.

raneq committed:
Add en_US translation as a diff reference

raneq committed:
Transform to poedit default format

Just opened file and save it without modifying anything by hand

raneq committed:
End catalan translation for emc_website

mclaeysb committed:
[FIX] emc: share transaction: product type and send functions

mclaeysb committed:
[REF] clean self vs for rec in self

mclaeysb committed:
[FIX] emc: fix transaction state widget

mclaeysb committed:
[FIX] emc: fix transaction email sending

mclaeysb committed:
[FIX] emc: fix transaction conversion

enricostano committed:
Replaced company_register_number with vat

mclaeysb committed:
[FIX] emc: cooperator_type depends

mclaeysb committed:
[FIX] emc: check share to_company and to_individual on transfer

houssine78 committed:
[IMP] make code a bit simpler.

houssine78 committed:
[FIX] fix flagging cooperator on share transfer

enricostano committed:
[FIX] coop_it_easy: Pick right email template for payment confirmation

robinkeunen committed:
[ADD] module readmes

houssine78 committed:
[FIX] fix wrong function call

houssine78 committed:
[FIX] take manuel comment into account.

houssine78 committed:
[IMP] remove duplicate act window overriding

robinkeunen committed:
[REF] OCA pre-commit compliance

robinkeunen committed:
[REF] compute method start with _compute

robinkeunen committed:
[REF] silence warning for sql injection and use of cr.commit()

robinkeunen committed:
automatic fixes

robinkeunen committed:
[REF] pretty xml

robinkeunen committed:
[REF] pretty js

robinkeunen committed:
[REF] redundant name module reference in xml_ids "account_cooperator_demo"

robinkeunen committed:
[REF] comment out unused reports

vvrossem committed:
[FIX] easy_my_coop: missing data_policy_approved ...
... causes installation crash for beesdoo_easy_my_coop

mclaeysb committed:
[ADD] easy_my_coop: internal_rules_approved and internal_rules_approved field on partner form

robinkeunen committed:
disable missing-manifest-dependency for res_partner_bank import

mclaeysb committed:
[ADD] easy_my_coop: effective date field on operation request

robinkeunen committed:
[REF] partner_age: remove emc dependency

houssine78 committed:
[FIX] fix xml view due to Odoo xml parsing glitch

houssine78 committed:
[IMP] do not return partner if company register number is None or empty

enricostano committed:
Add some catalan translation

robinkeunen committed:
[IMP] emc: get journal by xml id rather than code

robinkeunen committed:
[FIX] emc: add demo subscription account

and set it on the company to avoid
1) needing a localization for scratch install
2) bumbing into standard chart of account

robinkeunen committed:
[ADD] emca: subscription-request/id_/validate route

houssine78 committed:
[PORT] porting improvement from v9

from https://github.com/coopiteasy/vertical-cooperative/pull/93

cesarlr committed:
Added icon for menutitem and module description

robinkeunen committed:
[FIX] emca: add subscription demo journal

cesarlr committed:
Updated setuptools-odoo addons links

robinkeunen committed:
[ADD] emca: external_id_mixin

[IMP] emca: use external_id for product_template

houssine78 committed:
[IMP] hide if not taken in charge by emc

houssine78 committed:
[IMP] add internal rules approved field to view

houssine78 committed:
[IMP] add financial risk approval feature from dedicated PR.

https://github.com/coopiteasy/vertical-cooperative/pull/20/files

cesarlr committed:
pip install code labelled as shell

cesarlr committed:
Added description.rst to generate repository README (#102)

* https://github.com/coopiteasy/vertical-cooperative/pull/91

* Removed title and generated oca-gen

robinkeunen committed:
[FIX] emc: fix broken xml domains

mclaeysb committed:
[ADD] emc, emc_be, emc_taxshelter_report, emc_website, emc_website portal: fr and fr_be translations

mclaeysb committed:
[ADD] easy_my_coop: financial_risk_approved in partner view and subscription view

mclaeysb committed:
[REF] easy_my_coop_website: order Internal rules, Data policy, Financial risk

StCyr committed:
Replaces the address field by 2 fields, "street_name" and "house_number", in the "become a cooperator" web page.

These 2 new fields really only exist in this web page: Upon submission of the form, the address field is
computed and stored in the database, while the "street_name" and "house_number" fields are discarded.

robinkeunen committed:
make travis green again

mclaeysb committed:
[ADD] easy_my_coop: representative_of_member_company

robinkeunen committed:
[FIX] emc: do not use replace in views

houssine78 committed:
[IMP] add easy my coop flag on email template.

robinkeunen committed:
[FIX] emc: add invisible member to partner view

remytms committed:
[PEP] emc: Fix line length

robinkeunen committed:
make travis green again

cesarlr committed:
Fixed bad format date when no payment

mclaeysb committed:
[FIX] easy_my_coop: respresentatives view depends

danypr92 committed:
Make certificate email optional

danypr92 committed:
Make confirmation email optional

danypr92 committed:
Make capital release email optional

houssine78 committed:
[FIX] porting specific use case handling to v12

houssine78 committed:
[IMP] add import of some data from subscription request

houssine78 committed:
[FIX] little fix and hide some field in != use cases

danypr92 committed:
Name the argument for clarity when call to get_mail_template_notif

danypr92 committed:
Format the code to comply with the actual formating

danypr92 committed:
Add ROADMAP to easy_my_coop

robinkeunen committed:
[FIX] emc: test base

houssine78 committed:
[IMP] add a generic approval field to display on the subscription form

houssine78 committed:
[FIX] fix subscription request notification that was never sent

self in create function wasn't representing the record so the if
statement was always false

robinkeunen committed:
[REF] pre-commit semanticless fixes

robinkeunen committed:
[REF] split coop.py

remytms committed:
Revert "Replaces the address field by 2 fields, "street_name" and "house_number", in the "become a cooperator" web page."

This reverts commit a70dbfd6c640a27c46bb1cbf57a9967b694bccf6.

robinkeunen committed:
[REF] vincent bixolon evolution

danypr92 committed:
Add readonly to member and coop_candidate in Partner form

With these read only, we avoid creating a member without cooperator
number.

Fixes: #154

danypr92 committed:
Hide Create button from Coop Candidates and Members act windows

robinkeunen committed:
fixme out_inv_comm_type

danypr92 committed:
Move the readonly from member and coop_candidate to the model definition

We need this readonly in all the views of ResPartner.

victor-champonnois committed:
[IMP] easy_my_coop: copy=false for partner fields

Setting copy=False for member, cooperator and cooperator_register_number

victor-champonnois committed:
[IMP] easy_my_coop: don't display name on subscription request

Use firstname and lastname instead

Co-authored-by: robin.keunen <robin@keunen.net>

victor-champonnois committed:
[ADD] add a chatter for the subscription requests

victor-champonnois committed:
[IMP] remove "Delete access" for sub requests

victor-champonnois committed:
[ADD] button to archive a suscription request

victor-champonnois committed:
[IMP] redo subscription form from scratch

- refactor the form using templates
- simplify the design:
    - replace tables by bootstrap columns
    - use columns for the overall form rather than for individual fields
- use HTML5 widgets for the Part Numbers
- use adequate bootstrap classes
- use HTML5 date widget

victor-champonnois committed:
[ADD] easy_my_coop:  migration script : fill first/lastnames

carmenbianca committed:
[IMP] easy_my_coop: Ascertain that companies cannot request shares for individuals

And vice versa.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] easy_my_coop: Add dependency on partner_contact_gender

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] Adjust website URLs in all manifest files

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

robinkeunen committed:
[FIX] easy_my_coop: missing prefix in demo data

robinkeunen committed:
[ADD] emc_api: approval fields in subscription request services

robinkeunen committed:
[IMP] emc_api: restrict access to records managed by api clients

1st try: add an api user group and two ir.rules
- for api user group, domain: (1 = 1)
- for emc_user_group, domain: (external_id_generated = False)

This solution does not work because any group ir.rule bypasses
these rules, most notably, the rules from iwp to restrict access
to own structure.

This solution adds a global rules: if records were fetched or created
by actions resulting from api calls, no one can update or delete them.

The api bypasses these rules with .sudo.

A major drawback of this solution is that only Administrator will
be able to manipulate these records in the future. This may or may
not be good thing.

robinkeunen committed:
[FIX] pre-commit

huguesdk committed:
[REL] easy_my_coop 12.0.3.1.0

victor-champonnois committed:
[FIX] easy_my_coop: demo data for required firstname + lastname

[FIX] emc: fix demo data and tests for firstname

victor-champonnois committed:
[FIX] easy_my_coop: use firstname and lastname instead of name

- firstname in mail templates
- lastname in form breadcrumb (record name)

robinkeunen committed:
[REF] easy_my_coop: move send_capital_release_request to invoice

robinkeunen committed:
[REF] easy_my_coop: send waiting list email in function

robinkeunen committed:
[REF] easy_my_coop: all mails are configurable

victor-champonnois committed:
[IMP] set legal representative as true by default

robinkeunen committed:
[REF] easy_my_coop: extract send_confirmation_email

robinkeunen committed:
[REF] emc: consistent mail functions names

victor-champonnois committed:
[FIX] correct domain for partner views

- remove coop_candidates from cooperators kanban view
- keep only coop_candidates in coop_candidates view
- remove legal reprentative in both

robinkeunen committed:
[IMP] emc_api: update manages state change through methods

robinkeunen committed:
[FIX] tests

robinkeunen committed:
[ADD] emc_api: optional request fields

robinkeunen committed:
[REF] easy_my_coop: typo in word transfer*t*x

Not writing a migration script for field send_share_transfer_email
because it is not used yet in production.

robinkeunen committed:
[REF] easy_my_coop: consistent function names for sending mail

robinkeunen committed:
[REF] easy_my_coop: comment sudo use

robinkeunen committed:
[REF] easy_my_coop: extra quote in string

remytms committed:
[IMP] readme

victor-champonnois committed:
[FIX] representative only when parent company

robinkeunen committed:
[REF] easy_my_coop -> cooperator

robinkeunen committed:
[REF] easy_my_coop: demo data todo

robinkeunen committed:
[MIG] cooperator: set noupdate to false on report actions

huguesdk committed:
[REL] easy_my_coop 12.0.3.2.0

robinkeunen committed:
[NOP] easy_my_coop: todo comment: check subscription request fields

carmenbianca committed:
[FIX] Run pre-commit on all files (+ minor manual fixes)

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] Temporarily disable test

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

robinkeunen committed:
[FIX] easy_my_coop: split name and iban test

Commit:
Gender translation in subscription request selection

carmenbianca committed:
[FIX] Remove duplicate translation

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

Commit:
[UPD] Update all .pot files

robinkeunen committed:
[REF] cooperator : rename validated to is_valid_iban

- more explicit variable name
- test no longer check the id number validity
- fix tests

robinkeunen committed:
[FIX] coopeerator : re-enable successful tests

Commit:
[UPD] Update cooperator.pot

Commit:
[UPD] Update cooperator.pot

Commit:
[UPD] Update cooperator.pot

victor-champonnois committed:
[REF] replacing partner_age by partner_contact_birthdate

robinkeunen committed:
[REF] easy_my_coop_website_portal -> cooperator_portal

cesarlr committed:
Match between partner and sub_req done by vat

huguesdk committed:
[REL] easy_my_coop 12.0.3.3.0

robinkeunen committed:
[REF] easy_my_coop_payment_term -> cooperator

The translation files were not moved since the translated terms
were empty.

new field default_capital_release_request_payment_term made
not company dependent because company dependent fields don't
make sense on res.company model.

robinkeunen committed:
[REF] cooperator: property_cooperatror_account not company dependent

Company dependent fields don't make sense on res.company.

Also removed the required on the company configuration since
- it is caught in the code if empty.
- contraints on company fields without default cannot be set
  by odoo on installation.

robinkeunen committed:
[REF] cooperator: update readme

robinkeunen committed:
[REF] rename cooperator.subscription_request_form

wip

robinkeunen committed:
[REF] remove unused custom paperformat

robinkeunen committed:
[REF] cooperator: remove mention to easy my coop

robinkeunen committed:
[REF] rename action_invoice_tree_coop

robinkeunen committed:
[REF] rename account.journal xml_ids

robinkeunen committed:
[REF] rename cooperator_register_view.xml

robinkeunen committed:
[REF] rename mail_template xml ids

robinkeunen committed:
[REF] rename mail_template_view.xml

robinkeunen committed:
[REF] rename product xml ids

robinkeunen committed:
[REF] rename view_company_inherit_form2

robinkeunen committed:
[REF] rename partner xml ids

robinkeunen committed:
[REF] rename share product xml ids

carmenbianca committed:
[FIX] cooperator: Use openupgradelib-isms

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

robinkeunen committed:
[REF] remove cooperator/wizard/cooperative_history_wizard.xml

huguesdk committed:
[FIX] fix call to incorrect method name

fix incorrect call to ._get_share_transfer_mail_template() that was
forgotten during the recent correction of "transfert" to "transfer".

robinkeunen committed:
[REF] cooperator : rename menu ids

robinkeunen committed:
[REF] cooperator : rename group xml ids

robinkeunen committed:
[REF] cooperator: rename security rules

robinkeunen committed:
[REF] remove refarence to easy_my_coop in readme

robinkeunen committed:
[REF] cooperator: rename template xml ids

+ remove unused templates and css classes

robinkeunen committed:
[REF] cooperator: remove unused css classes

robinkeunen committed:
[REF] cooperator: remove mentions to easy_my_coop in js files

robinkeunen committed:
[REF] cooperator: move cooperator.js to easy_my_coop_website

robinkeunen committed:
[REF] cooperator: rename easy_my_coop_data.xml

robinkeunen committed:
[REF] cooperator: rename field easy_my_coop on mail.template

robinkeunen committed:
[REF] cooperator: rename easy_my_coop_report.xml

huguesdk committed:
[FIX] fix subscription request name field

* make subscription.request.name a stored computed field to allow to
  search by any part of the full name.
* fix subscription request views: use only firstname and lastname.

Commit:
[UPD] Update easy_my_coop.pot

robinkeunen committed:
[REF] cooperator: remove mentions to easy my coop in tests

robinkeunen committed:
[FIX] cooperator: fix migration scripts to 12.0.5.0.0

carmenbianca committed:
[FIX] easy_my_coop: Fix name computation

Previously, if either firstname or lastname was (unexpectedly) empty, an
error would be thrown.

This commit uses approximately the same code approach as
partner-contact/partner_firstname to compute the full name.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] easy_my_coop: Improve English source strings

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

Commit:
[UPD] Update easy_my_coop.pot

huguesdk committed:
[FIX] fix create company contact with sub-contact

fix error that happens when creating a company contact with a
sub-contact.

huguesdk committed:
[FIX] fix create subscription request from partner

fix creation of subscription request from partner by adding missing
fields: firstname, lastname and lang.

Commit:
[UPD] Update easy_my_coop.pot

victor-champonnois committed:
[REF] rename easy_my_coop_website: cooperator

Commit:
[UPD] Update easy_my_coop.pot

carmenbianca committed:
[FIX] cooperator: Do not search account 416000

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

robinkeunen committed:
[REF] minor refactoring

victor-champonnois committed:
[UPD] translation : cooperator

victor-champonnois committed:
[IMP] Add new translations : cooperator

carmenbianca committed:
[REM] cooperator: Remove duplicate translation

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[UPD] Update cooperator.pot

Commit:
[UPD] Update cooperator.pot

Commit:
[UPD] Update cooperator.pot

victor-champonnois committed:
run pre-commit

Commit:
[UPD] Update cooperator.pot

victor-champonnois committed:
[FIX] duplicated dependency in cooperator

carmenbianca committed:
[FIX] easy_my_coop: Fix name computation crash

Don't use value from self when looping over a list

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

Commit:
[UPD] Update cooperator.pot

github-grap-bot committed:
[UPD] changelog

github-grap-bot committed:
[UPD] README.rst

github-grap-bot committed:
easy_my_coop 12.0.3.3.2

github-grap-bot committed:
[UPD] README.rst

carmenbianca committed:
[FIX] SCRLfs → SC

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

huguesdk committed:
[MIG] cooperator: migration to 13.0

huguesdk committed:
[MIG] cooperator: migration to 14.0

huguesdk committed:
[IMP] run pre-commit

carmenbianca commented:
You can disable eslint. Related to #227

huguesdk committed:
[FIX] fix company subscription request validation

carmenbianca committed:
[FIX] Add .eslintignore

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[REM] cooperator: dependency on email_template_config

email_template_config adds a field `force_email_send` to mail templates.
This field is referenced nowhere in the cooperator code.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca commented:
Did a little bit of tinkering on this to get the tests to run.

The removal of dependencies `email_template_config` and `partner_contact_address` merits discussion at CC, so I've added it to the agenda.

@victor-champonnois I think you're working on this this week. I believe the following must be done:

1. Repair failing tests.
2. Do a tiny little bit of functional testing. If some things don't work, consider writing automated tests for them.
3. Rebase other v14 PRs on top of this branch, and set `14.0-mig-cooperator` as the target branch of those PRs. Their tests should begin running, and you should be able to do functional testing on those branches.
4. Request code reviews once the tests are green (or almost green). Also poke the analysts to see how/if they want to do functional testing.

carmenbianca commented:
`partner_contact_address` is a functional requirement of `cooperator`. Quoting Cath:

> This is useful => when you take a share as a company, you legally have to enter a representative. 
>
> I've put some video of this on the cooperator task
>
> [...]
>
> we need the representative on cooperator, that"s all I know

victor-champonnois committed:
[FIX] session values when creating partner via form

victor-champonnois committed:
[ADD] legal_form to partner form

victor-champonnois commented:
Adding a field in partner form, following the discussion on [this task](https://gestion.coopiteasy.be/web#id=8756&model=project.task&view_type=form&menu_id=) (@carmenbianca before you force push ^^⁾

victor-champonnois committed:
[IMP] generic rule translation to distinguish from ROI

victor-champonnois committed:
[FIX] company reference in mail_template

victor-champonnois committed:
[REM] old_member from cooperator tree view

victor-champonnois committed:
[REM] update_partner_info wizard

victor-champonnois committed:
[REM] create button for subscription register

victor-champonnois committed:
[IMP] replace subrequest type subscription with new

victor-champonnois committed:
[REM] fr_be translation

There is actually no difference between fr and fr_be.

victor-champonnois committed:
[UPD] Update cooperator.pot

victor-champonnois committed:
[UPD] fr translations

victor-champonnois committed:
[IMP] fix typo and add trad to mail templates

carmenbianca committed:
[REF] cooperator: Create get_company_type_selection

If we don't do this, Odoo will complain that we should be using
selection_add.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

huguesdk committed:
[FIX] fix view inheritance

cooperator.view_res_partner_filter view references the supplier field,
which is defined in account.res_partner_view_search and not in
base.view_res_partner_filter.

huguesdk committed:
[IMP] improve fr translation

huguesdk committed:
[FIX] add capital release request report binding

allow to generate capital release request reports from the print menu.

huguesdk committed:
[IMP] increase fr translation

huguesdk committed:
[FIX] fix subscription journal sequence format

huguesdk committed:
[FIX] pass pre-commit

huguesdk committed:
[FIX] add test-requirements.txt

huguesdk committed:
[FIX] fix company reference in report

fix company reference in cooperator certificate report.

huguesdk committed:
[FIX] fix email templates

* remove newline characters in subjects; newline characters in subjects causes
  an error which prevents messages from being sent.
* remove a period at the end of a subject.
* replace "certificat" by "certificate" in report names.

carmenbianca commented on cooperator/models/account_move.py:
```diff
@@ -0,0 +1,208 @@
+            "type": "subscription",
+        }
+
+    def get_membership_vals(self):
+        # flag the partner as an effective member
+        # if not yet cooperator we generate a cooperator number
+        vals = {}
+        if self.partner_id.member is False and self.partner_id.old_member is False:
+            sequence_id = self.get_sequence_register()
+            sub_reg_num = sequence_id.next_by_id()
+            vals = {
+                "member": True,
+                "old_member": False,
+                "cooperator_register_number": int(sub_reg_num),
+            }
+        elif self.partner_id.old_member:
+            vals = {"member": True, "old_member": False}
+
+        return vals
```
This seems incredibly error-prone to me.

1. Method is named `get_*`, but in fact changes the state of the sequence register.
2. No explicit check whether `cooperator_register_number` already has a value. The check is implicit through the value of `old_member`.

Is there a better place to set `cooperator_register_number`?

huguesdk committed:
[FIX] correct report template source strings

huguesdk committed:
[FIX] fix report attachment names

fix capital release request report attachment names: base the value on
the default invoice attachment names, which results in the same value
for the 3 file types (unless the name is not set or the state is not
"posted"): internal account.move attachment, email attachment and print
report file name.

huguesdk committed:
[IMP] improve fr translation

* improve french translation (by checking the differences between the
  old fr and fr_be translations).
* add fr_be translation for specific belgian terms only.

carmenbianca commented on cooperator/models/operation_request.py:
```diff
@@ -114,40 +113,34 @@ def _compute_subscription_amount(self):
     )
 
-    invoice = fields.Many2one("account.invoice", string="Invoice")
+    invoice = fields.Many2one("account.move", string="Invoice")
```
Good opportunity to rename to `invoice_id` maybe. Or `invoice_move_id`? Not sure.

huguesdk replied:
indeed, it should be renamed. there are several other things that need renaming, and my understanding was that we will first do all of these in 12.0 as part of komunigi and then port them to 14.0. not doing it as part of the migration allows to reduce the probability of new bugs not directly related to the migration.

carmenbianca reviewed (approved):
one comment, but LGTM.

massive work. thanks @huguesdk !

huguesdk commented:
/ocabot merge nobump
